### PR TITLE
Move XPU mean and mean_grad to phi

### PR DESCRIPTION
**Pull request by 0x45f**
### PR types

Others
### PR changes

Others
### Describe

Move XPU mean and mean_grad to phi

### CODE REVIEW

File: paddle/phi/kernels/xpu/mean_all_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  std::vector<int> x_shape;
+  x_shape.push_back(1);
+  x_shape.push_back(input->numel());
+  std::vector<int> rdims = {1};
+  int r = xpu::reduce_mean(dev_ctx.x_context(),
+                           reinterpret_cast<const XPUType*>(x_data),
+                           reinterpret_cast<XPUType*>(y_data),
+                           x_shape,
+                           rdims);
+  PADDLE_ENFORCE_EQ(
```
Comment (helen88) [inline paddle/phi/kernels/xpu/mean_all_kernel.cc:42]: PADDLE_ENFORCE_XDNN_SUCCESS 请用这个宏

File: paddle/phi/kernels/xpu/mean_all_grad_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+template <typename T, typename Context>
+void MeanAllGradKernel(const Context& dev_ctx,
+                       const DenseTensor& x,
+                       const DenseTensor& out_grad,
+                       DenseTensor* x_grad) {
+  using XPUType = typename XPUTypeTrait<T>::Type;
+
+  auto OG = &out_grad;
+  PADDLE_ENFORCE_EQ(
```
Comment (helen88) [inline paddle/phi/kernels/xpu/mean_all_grad_kernel.cc:31]: PADDLE_ENFORCE_XDNN_SUCCESS 请用这个宏，下同

File: paddle/phi/kernels/xpu/mean_all_kernel.cc (hunk 1 above)
Comment (0x45f) [reply paddle/phi/kernels/xpu/mean_all_kernel.cc]: 两处宏已经修改，感谢

**Review by helen88 — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]

**Pull request merged**